src/modules/user/controllers/login/login.controller.ts

```diff
@@ -0,0 +1,45 @@
+import { type NextFunction, type Request, type Response } from 'express';
+import { UserService } from '../../index';
+import EncryptionService from '../../../../services/EncryptionService';
+import TokenService from '../../../../services/TokenService';
+import { type IUser } from '../../type';
+import { NotAuthorizedError } from '../../../../errors';
+import { SessionService } from '../../../session';
+import { v4 as uuidv4 } from 'uuid';
+
+const login = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+  const { email, password } = req.body;
+
+  try {
+    const user: IUser | null = await UserService.getUserByEmail(email);
+
+    if (!user) {
+      throw new NotAuthorizedError('User could not be found');
+    }
+
+    const isPasswordEqual = await EncryptionService.comparePassword(password, user.password);
+
+    if (!isPasswordEqual) {
+      throw new NotAuthorizedError('Wrong password');
+    }
```

Comment:
this code should be implemented in UserService.getUserByPasswordAndEmail

Reply:
and better to use passport.basicAuth/localAuth for finding user by email/psw and assigning it to request context